.github/pull_request_template.md

```diff
@@ -0,0 +1,33 @@
+### Description
+What does this PR do?
+
+### TODO if you're a community member
+- [ ] Make sure there's a changelog. This is a description of what changed from
+  a user's point of view.
+
+### TODO if you're a HashiCorp employee
+- [ ] Make sure there's a changelog. Brand new features have a differently
```

Comment:
I think the changelog check should obviate the need for this checkbox. It's worth noting that not all changes require changelogs, either.

I also don't know that this is always true?
```
If this
  PR is the CE portion of an ENT PR, put the changelog here, _not_ in your ENT
  PR.
```

If it's for an enterprise feature, my understanding is that the changelog goes in the ent PR so it signals it's for ent only. For example, if an update to lease count quotas has a CE change, the changelog should be on the ent PR.

Reply:
The official guidance doesn't speak to changes that are purely ent-only, but is:

> If you are making a change for an issue and it requires an enterprise PR and a CE PR, it’s preferable to make the changelog entry in the CE repository with the changelog entry name corresponding to the CE PR number, not the enterprise PR number. Having the changelog entry in the CE repo lets us link to the CE PR, whereas in the enterprise repo we cannot link to the PR.

Reply:
Oh, huh, TIL. I guess I've always been doing it counter to that advice for features that are primarily in ent

Reply:
I like including this here because although yes, there is a GHA check for a changelog, the GHA check does not remind you that features need to have a changelog that is formatted differently than other changelogs, something that myself and others have gotten dinged on in the past.

Reply:
Could we move that information to the GHA instead? It should just be updating this bit, which I think is a little outdated compared to the `contributing.md` link:

```
            if [ -z "$toolchain_files" ]; then
              echo "Not found."
              echo ""
              echo "Did not find a changelog entry named ${expected_changelog_file}"
              echo "If your changelog file is correct, skip this check with the 'pr/no-changelog' label"
              echo "Reference - https://github.com/hashicorp/vault/pull/10363 and https://github.com/hashicorp/vault/pull/11894"
              exit 1
            fi
```

Reply:
That will only show up in GHA if there's a problem in the changelog though, no? If I include a changelog with my PR, the GHA check is satisfied and nothing is logged, even if my changelog is incorrectly formatted.

Reply:
That would show up if there isn't a changelog, yeah. You're right, that wouldn't show up if I'd incorrectly formatted a feature, but we do have a lot of checks for incorrectly formatted changelogs there today -- it'd fit neatly. My bash-fu isn't great but it'd probably be something like this:

```
          elif grep -q ':feature$' "$changelog_files"; then
            # Pseudocode: if second line doesn't start with **
            echo "Incorrect feature formatting, make sure you do X instead"
            exit 1
```

If you'd like, I can play around with this a little and see if I can get something working. I've also done this incorrectly in the past and it'd be great to add this to the list of things we check. If you're not feeling up to doing this now then I don't think it should block the PR template, but it'd be great if we could automate that part of the checklist away

Reply:
It's not something I'm going to focus on in this PR, since this is really just about the template, but if you want to follow this with a PR of your own that automates a formatting check and removes an item from the checklist, that sounds great to me.

To be clear, I'm not a checklist enthusiast. In my happy world, we wouldn't have checklists at all and automation would handle literally everything. But I think this is better than nothing at all until we get to that point.

Maybe if enough people are irritated with the checklist, it will spur the creation of additional automation 😅

Reply:
Totally agreed! This was useful to understand the root of the desire for the checklist here, and knowing that you'd be happy if we automated it is great news to me. Checklists are easy, effective and give us a great list of things that we could automate to make our lives even easier